### ISSUE TITLE
iOS caller Id and notifications

### ISSUE THREAD

**Issue by RUIFERNANDE5** [hoxfon/react-native-twilio-programmable-voice#48]
What version of React Native are you running? 0.49
What version of react-native-twilio-programmable-voice are you running? ^3.2.0
What device are you using? (e.g iOS9 simulator, Android 6 device)? iOS 11 device

Hey, I'm currently able to make and receive a call with CallKit configured. However digging the docs and even the code I was not able to understand how can I change the received caller name. FE: I receive a call from twilio by the user "Rui", however to receive this callerId as a string I have to prepend "client:" to the string, or the caller Id will treat it like a number and convert the name "Rui" to  "784". 
The thing is, with "client:Rui" on the twilio server as callerId the iOS call screen will display: "client:Rui", my question is, how can I trim this "client:"?
Also, can't I just display a call notification instead of the popup caller screen? So when I press the notification it'll open the app with my custom screen.

P.S: I have limited understanding of ObjC / Java, I'm kinda struggling atm
Thanks for the great package! Also, your app deserves way more downloads.

**Comment by fabriziomoscon** [comment]
Hi Rui, sorry for my delayed answer.
[...] `how can I change the received caller name` => that is the call `callerId` right?
Could you try this TwiML and tell me how it goes?

```
<Response>
	<Dial callerId="Rui" [...]>// callerId must be a valid number of a client identifier 
		<Client [...]>Rui</Client> // this Rui is the client identifier to call, the part after the `client:`
	</Dial>
</Response>
```

From the docs: `The client identifier currently may only contain alpha-numeric and underscore characters.`
https://www.twilio.com/docs/api/twiml/dial
https://www.twilio.com/docs/api/twiml/client

In regard to your second question about letting your app take control over incoming calls and preventing callkit to handle them I was not able to do it so far. Please let me know if you find a way.

**Comment by rclai** [comment]
The `callerId` on that `Dial` verb cannot be whatever, it must be a valid number or a client id.

What needs to happen is we need to somehow be able to change the `localizedCallerName` on the `CXCallUpdate` object when the call comes in. However, it would be ideal if this can be invoked on the javascript side because that's probably where the data needed to set it is.

It would be nice if that `reportNewIncomingCallWithUUID` was controlled on the JS side like this other CallKit library does it https://github.com/ianlin/react-native-callkit#displayincomingcall

**Comment by langan** [comment]
Hi @rclai & @RUIFERNANDE5 

I am having the exact same issue, did you ever find a solution?

**Comment by lonnylot** [comment]
The original issue duplicates #89 and is fixed in #137 

I agree with @rclai, but as it's a separate issue and we need to find a solution for Android I have opened a separate issue specific for this #139

**Comment by jdegger** [comment]
I'm closing this issue in favor of #89.

Read our repository update in #158. I invite you to reopen this issue when you still have problems so we can debug the problems together.
